src/index.ts

```diff
@@ -34,9 +34,22 @@ async function run() {
 
   core.debug(`Changed files: ${changedFiles}`)
 
+  let extensions = core.getInput("extensions").split(" ")
+  core.debug(`Extensions: ${extensions}`)
+  let changedFilesMatchingExtensions = changedFiles.filter((file) =>
+    extensions.some((ext) => file.endsWith(ext)),
+  )
+  core.debug(
+    `Changed files matching extensions: ${changedFilesMatchingExtensions}`,
+  )
```

Comment:
I'm going back and forth a bit on the manual filtering of files.  For older versions of ESLint, we need to support something, but the translation of these `extensions` to the `--ext` CLI option seems a straightforward path there.  If they're on a newer version of ESLint that supports the `files` config, can we push teams in the direction of using that in their own app?  

Other than it requiring a bit of work, are there any negative consequences for the teams?  It seems like we'd be pushing them in a more sustainable future for themselves too (i.e. any ESLint run in any other environment will be constrained just the same as their Balto ESLint runs are).

Reply:
@molawson, thanks for bringing this up. I felt a bit uneasy too when working on it, but couldn't immediately think of a better solution. After looking in more to follow up on your comments I realized: this is a problem that I forgot that I already solved 😆

ESLint 9 will let you throw whatever junk you want in the file args and will filter it based off of the `files` option in the `eslint.config.js` file and the built-in defaults:

https://github.com/eslint/eslint/blob/17cfb684194df48d0a5dc566af9c28fe80ea6d42/lib/config/default-config.js#L64-L74

As long as you pass `—no-warn-ignore` it won't even complain that there are files that don't match.

I added that flag earlier this year when spiking out the "clean sheet" version for v9. A few weeks ago I pulled that flag out to maximize support for earlier versions, but now it's clear we need to add forking logic for below and above v9.

Reply:
I just got off a call with @molawson. I looked into doing the `ext`/`files` config split today and got far with it, but ran into issues with older versions of ESLint not following the `ext` flag the way we thought it would. Since we previously manually filtered files in v0, we decided it would be good to move forward with this current implementation (after making the space -> comma separated list change).